tools/generate_features.py

```diff
@@ -88,174 +86,142 @@ def drop_close_bright_stars(
-                            "filter": {"phot_g_mean_mag": {"$lt": 13.0}},
-                            "projection": {
-                                "phot_g_mean_mag": 1,
-                                "bp_rp": 1,
-                                "coordinates.radec_geojson.coordinates": 1,
-                            },
-                        }
-                    },
-                    "filter": {},
-                },
-            }
+    q = kowalski_instance.query(query)
 
-            q = kowalski_instance.query(query)
-            gaia_results = q['data'][catalog]
-            gaia_results_dct.update(gaia_results)
+    gaia_results = q.get('data')
+    gaia_results_dct.update(gaia_results['Gaia_EDR3'])
 
     print('Identifying sources too close to bright stars...')
+
     # Loop over each id to compare with query results
-    count = 0
-    for id in ids:
-        count += 1
-        if count % limit == 0:
-            print(f"{count} done")
-        if count == len(ids):
-            print(f"{count} done")
+    query_result = np.array(gaia_results_dct['quad'])
 
-        val = id_dct[id]
-
-        ra_geojson, dec_geojson = val['radec_geojson']['coordinates']
-
-        single_result = gaia_results_dct[str(id)]
-        if len(single_result) > 0:
-            coords = np.array(
-                [
-                    x['coordinates']['radec_geojson']['coordinates']
-                    for x in single_result
-                ]
-            )
-            coords[:, 0] += 180.0
-
-            # SkyCoord object for query results
-            Coords = SkyCoord(coords, unit=['deg', 'deg'])
-            # SkyCoord object for input source
-            coord = SkyCoord(ra_geojson + 180.0, dec_geojson, unit=['deg', 'deg'])
-
-            all_separations = Coords.separation(coord)
-            # Identify closest source to input
-            drop_source = np.argmin(all_separations)
-
-            # If closest source is within specified radius, treat it as the input source and drop it from further consideration
-            xmatch_source = {}
-            if all_separations[drop_source] < xmatch_radius_arcsec * u.arcsec:
-                xmatch_source = single_result.pop(drop_source)
-                Coords = np.delete(Coords, drop_source)
-
-            # If possible, use all-Gaia coordinates for next step
-            if len(xmatch_source) > 0:
-                xmatch_ra, xmatch_dec = xmatch_source['coordinates']['radec_geojson'][
-                    'coordinates'
-                ]
-                xmatch_ra += 180.0
-                xmatch_coord = SkyCoord(xmatch_ra, xmatch_dec, unit=['deg', 'deg'])
-            else:
-                xmatch_coord = SkyCoord(
-                    ra_geojson + 180.0, dec_geojson, unit=['deg', 'deg']
-                )
+    if len(query_result) > 0:
+        source_coords = np.array(
+            [x['radec_geojson']['coordinates'] for x in id_dct.values()]
+        )
+        source_coords[:, 0] += 180.0
```

Comment:
Should this be np.mod()?

Reply:
This line and line 162 address the [-180, 180] range of geojson RA values. I think it should stay as-is - an added constant to get to a [0, 360] range.